src/libraries/System.Net.Quic/src/System.Net.Quic.csproj

```diff
@@ -191,11 +174,11 @@
                           '$(Configuration)' == 'Debug' and
                           '$(DotNetBuildSourceOnly)' != 'true'">
       <!-- MsQuic packages do not have macOS binaries. Our transport package is only one source. -->
-      <PackageReference Include="runtime.osx-$(TargetArchitecture).runtime.native.System.Net.MsQuic.Transport"
-                        Version="$(SystemNetMsQuicTransportVersion)"
+      <PackageReference Include="MsQuic.Native"
```

Comment:
This is the main change consuming the newest MsQuic, manually uploaded to our internal NuGet feed.

The rest is docs update and cleanup of the now unused transport package consumption.